scripts/generate-indexer-config.js

```diff
@@ -1,41 +1,101 @@
 const dotenv = require("dotenv");
-const path = require("path")
-const fs = require("fs")
+const path = require("path");
+const fs = require("fs");
+const http = require("http");
+const crypto = require("crypto");
 
-const envPath = path.join(__dirname, "../packages/api-server/.env")
+const envPath = path.join(__dirname, "../packages/api-server/.env");
 
-dotenv.config({path: envPath})
+dotenv.config({ path: envPath });
 
-const wsRpcUrl = process.argv[2];
+function sendJsonRpc(rpcUrl, method, params) {
+  const url = new URL(rpcUrl);
 
-let config = {
-  l2_sudt_type_script_hash: process.env.L2_SUDT_VALIDATOR_SCRIPT_TYPE_HASH,
-  polyjuice_type_script_hash: process.env.POLYJUICE_VALIDATOR_TYPE_HASH,
-  rollup_type_hash: process.env.ROLLUP_TYPE_HASH,
-  eth_account_lock_hash: process.env.ETH_ACCOUNT_LOCK_HASH,
-  godwoken_rpc_url: process.env.GODWOKEN_JSON_RPC,
-  pg_url: process.env.DATABASE_URL,
-  chain_id: process.env.CHAIN_ID,
-  sentry_dsn: process.env.SENTRY_DNS,
-  sentry_environment: process.env.SENTRY_ENVIRONMENT,
-}
+  return new Promise((resolve, reject) => {
+    const data = JSON.stringify({
+      id: crypto.randomUUID(),
+      jsonrpc: "2.0",
+      method: method,
+      params: params,
+    });
+
+    const options = {
+      hostname: url.hostname,
+      port: url.port,
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "Content-Length": data.length,
+      },
+    };
+
+    const req = http.request(options, (res) => {
+      res.on("data", (d) => {
+        const res = JSON.parse(d.toString());
+        if (res.error) {
+          reject(res);
+        } else {
+          resolve(res.result);
+        }
+      });
+    });
 
-if (wsRpcUrl) {
-  config.ws_rpc_url = wsRpcUrl;
+    req.on("error", (error) => {
+      console.error(error);
+    });
+
+    req.write(data);
+    req.end();
+  });
 }
 
-let tomlStr = "";
+const run = async () => {
+  const nodeInfo = await sendJsonRpc(
```

Comment:
```suggestion
  const nodeInfo = await new RPC(process.env.GODWOKEN_JSON_RPC).gw_get_node_info();
```

Reply:
I prefer to use the `sendJsonRpc` method since the scripts here might not have node_module installed with RPC module, scripts are outer tool for generic usage.